logfire/_internal/main.py

```diff
@@ -1698,7 +1698,17 @@ def message_template(self) -> str | None:  # pragma: no cover
 
     @property
     def tags(self) -> Sequence[str]:  # pragma: no cover
-        return self._get_attribute(ATTRIBUTES_TAGS_KEY, [])
+        return self._get_attribute(ATTRIBUTES_TAGS_KEY, ())
+
+    @tags.setter
+    @handle_internal_errors()
+    def tags(self, new_tags: Sequence[str]) -> None:
+        """Set or add tags to the span."""
+        updated_tags = uniquify_sequence(list(new_tags))
+        if self._span is None:  # pragma: no cover
```

Comment:
```suggestion
        if self._span is None:
```

Please test this case by setting tags before `with span:`